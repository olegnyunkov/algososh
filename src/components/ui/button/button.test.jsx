import React from "react";
import renderer from "react-test-renderer";
import {Button} from "./button";

describe("button rendered: ", () => {

  it("without text", () => {
    const btn = renderer.create(<Button />).toJSON();
    expect(btn).toMatchSnapshot();
  });

  it("with text", () => {
    const btn = renderer.create(<Button text={"Кнопка"}/>).toJSON();
    expect(btn).toMatchSnapshot();
  });

  it("with disabled state", () => {
    const btn = renderer.create(<Button disabled={true}/>).toJSON();
    expect(btn).toMatchSnapshot();
  });

  it("with loading state", () => {
    const btn = renderer.create(<Button isLoader={true}/>).toJSON();
    expect(btn).toMatchSnapshot();
  });
})
