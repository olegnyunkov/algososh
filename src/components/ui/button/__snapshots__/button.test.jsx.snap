// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button rendered:  with disabled state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button rendered:  with loading state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button rendered:  with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`button rendered:  without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
